ccl/changefeedccl: TestCatchupScanOrdering failed

ccl/changefeedccl.TestCatchupScanOrdering [failed](https://tanzanite.cluster.engflow.com/invocations/default/b2ffac8e-5aa4-47b6-beaf-ce36c8a7d3d8?testReportRun=3&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [603ca2299bf61260492cbe544d5caffc414d8d54](https://github.com/cockroachdb/cockroach/commits/603ca2299bf61260492cbe544d5caffc414d8d54):


```
    validations_test.go:71: time taken to make the 6296-th bank transfer: 35.34089ms
    validations_test.go:71: time taken to make the 6297-th bank transfer: 45.69017ms
    validations_test.go:71: time taken to make the 6298-th bank transfer: 40.859301ms
    validations_test.go:71: time taken to make the 6299-th bank transfer: 52.789581ms
    validations_test.go:71: time taken to make the 6300-th bank transfer: 52.14131ms
    validations_test.go:71: time taken to make the 6301-th bank transfer: 74.527082ms
    validations_test.go:71: time taken to make the 6302-th bank transfer: 84.992971ms
    validations_test.go:71: time taken to make the 6303-th bank transfer: 56.357931ms
    validations_test.go:71: time taken to make the 6304-th bank transfer: 44.38484ms
    validations_test.go:71: time taken to make the 6305-th bank transfer: 38.631491ms
    validations_test.go:71: time taken to make the 6306-th bank transfer: 45.396881ms
    validations_test.go:71: time taken to make the 6307-th bank transfer: 33.37094ms
    validations_test.go:71: time taken to make the 6308-th bank transfer: 35.068221ms
    validations_test.go:71: time taken to make the 6309-th bank transfer: 27.2631ms
    validations_test.go:71: time taken to make the 6310-th bank transfer: 27.48738ms
    validations_test.go:71: time taken to make the 6311-th bank transfer: 22.668671ms
    validations_test.go:71: time taken to make the 6312-th bank transfer: 30.29951ms
    validations_test.go:71: time taken to make the 6313-th bank transfer: 23.162621ms
    validations_test.go:71: time taken to make the 6314-th bank transfer: 33.646081ms
    validations_test.go:71: time taken to make the 6315-th bank transfer: 28.423121ms
    validations_test.go:71: time taken to make the 6316-th bank transfer: 39.744261ms
    validations_test.go:71: time taken to make the 6317-th bank transfer: 44.927711ms
    validations_test.go:71: time taken to make the 6318-th bank transfer: 45.17037ms
    validations_test.go:71: time taken to make the 6319-th bank transfer: 37.367421ms
    validations_test.go:71: time taken to make the 6320-th bank transfer: 40.59008ms
    validations_test.go:71: time taken to make the 6321-th bank transfer: 26.187411ms
    validations_test.go:71: time taken to make the 6322-th bank transfer: 20.68922ms
    validations_test.go:71: time taken to make the 6323-th bank transfer: 19.69265ms
    validations_test.go:71: time taken to make the 6324-th bank transfer: 23.947181ms
    validations_test.go:71: time taken to make the 6325-th bank transfer: 39.77697ms
    validations_test.go:71: time taken to make the 6326-th bank transfer: 30.044561ms
    validations_test.go:71: time taken to make the 6327-th bank transfer: 40.88427ms
    validations_test.go:71: time taken to make the 6328-th bank transfer: 48.92272ms
    validations_test.go:71: time taken to make the 6329-th bank transfer: 33.721371ms
    validations_test.go:71: time taken to make the 6330-th bank transfer: 43.52647ms
    validations_test.go:71: time taken to make the 6331-th bank transfer: 37.876001ms
    validations_test.go:71: time taken to make the 6332-th bank transfer: 40.39085ms
    validations_test.go:71: time taken to make the 6333-th bank transfer: 28.513481ms
    validations_test.go:71: time taken to make the 6334-th bank transfer: 23.59431ms
    validations_test.go:71: time taken to make the 6335-th bank transfer: 32.646491ms
    validations_test.go:71: time taken to make the 6336-th bank transfer: 28.8967ms
    validations_test.go:71: time taken to make the 6337-th bank transfer: 33.427221ms
    validations_test.go:71: time taken to make the 6338-th bank transfer: 34.22127ms
    validations_test.go:71: time taken to make the 6339-th bank transfer: 41.269841ms
    validations_test.go:71: time taken to make the 6340-th bank transfer: 25.16394ms
    validations_test.go:71: time taken to make the 6341-th bank transfer: 34.765421ms
    validations_test.go:71: time taken to make the 6342-th bank transfer: 28.295201ms
    validations_test.go:71: time taken to make the 6343-th bank transfer: 31.671001ms
    validations_test.go:71: time taken to make the 6344-th bank transfer: 33.90339ms
--- FAIL: TestCatchupScanOrdering/cloudstorage/bank (303.50s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCatchupScanOrdering.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51674

--- a/pkg/ccl/changefeedccl/validations_test.go
+++ b/pkg/ccl/changefeedccl/validations_test.go
@@ -14,7 +14,6 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest"
-	"github.com/cockroachdb/cockroach/pkg/ccl/utilccl"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -26,12 +25,12 @@ import (
 func TestCatchupScanOrdering(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	defer utilccl.TestingEnableEnterprise()()
 
 	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
 		t.Run("bank", func(t *testing.T) {
 			ctx := context.Background()
 			const numRows, numRanges, payloadBytes, maxTransfer = 10, 10, 10, 999
+			const checkRows = 200
 			gen := bank.FromConfig(numRows, numRows, payloadBytes, numRanges)
 			var l workloadsql.InsertsDataLoader
 			if _, err := workloadsql.Setup(ctx, s.DB, gen, l); err != nil {
@@ -62,7 +61,7 @@ func TestCatchupScanOrdering(t *testing.T) {
 			g.GoCtx(func(ctx context.Context) error {
 				prevTimeTransfer := time.Now()
 				for {
-					if atomic.LoadInt64(&done) > 0 {
+					if atomic.LoadInt64(&done) > 0 || numOfTrans.Load() > 2*checkRows {
 						return nil
 					}
 					if err := randomBankTransfer(numRows, maxTransfer, s.DB); err != nil {
@@ -95,7 +94,7 @@ func TestCatchupScanOrdering(t *testing.T) {
 					t.Logf("key: %s, value: %s\n", m.Key, m.Value)
 					t.Logf("time taken to see the %d-th changefeed change: %v", seenChanges, time.Since(prevTimeCDC))
 					prevTimeCDC = time.Now()
-					if seenChanges >= 200 {
+					if seenChanges >= checkRows {
 						atomic.StoreInt64(&done, 1)
 						break
 					}

